app/views/collections/show.rb

```diff
@@ -176,5 +176,11 @@ def preview_search_data
         }
       end
     end
+
+    def facet_entry_field_title(facet_entry)
```

Comment:
Could we reduce repeated execution here with something like:
```ruby
@facet_entry_field_titles ||= {}
@facet_entry_field_titles[facet_entry.facet_field] ||= begin
  ff = Europeana::Blacklight::Response::Facets::FacetField.new(facet_entry.facet_field, [])
  presenter = FacetPresenter.build(ff, controller)
  presenter.facet_title || facet_entry.facet_field
end
```
i.e. so that the instantiation of the more complex objects only occurs once for each facet field.